src/main/java/org/apache/commons/compress/archivers/dump/DumpArchiveInputStream.java

```diff
@@ -547,7 +550,8 @@ private void readDirectoryEntry(DumpArchiveEntry entry) throws IOException {
 
                 final byte type = blockBuffer[i + 6];
 
-                final String name = DumpArchiveUtil.decode(zipEncoding, blockBuffer, i + 8, blockBuffer[i + 7]);
+                final String name =
+                        DumpArchiveUtil.decode(zipEncoding, blockBuffer, i + 8, Byte.toUnsignedInt(blockBuffer[i + 7]));
```

Comment:
This change fixes an unrelated type conversion problem. I created a separate PR #711 for this, but I need the fix for the unit tests to succeed.